holoviews/util/__init__.py

```diff
@@ -484,11 +485,18 @@ def builder(cls, spec=None, **kws):
             return Options(spec, **kws)
 
         filtered_keywords = [k for k in completions if k not in cls._no_completion]
-        kws = ', '.join('{opt}=None'.format(opt=opt) for opt in sorted(filtered_keywords))
-        builder.__doc__ = '{element}({kws})'.format(element=element, kws=kws)
+        sorted_kw_set = sorted(set(filtered_keywords))
+        kws = ', '.join('{opt}=None'.format(opt=opt) for opt in sorted_kw_set)
+        if sys.version_info.major == 2:
+            builder.__doc__ = '{element}({kws})'.format(element=element, kws=kws)
+        else:
+            signature = Signature([Parameter('cls', Parameter.POSITIONAL_ONLY)]
+                                  + ([Parameter('spec', Parameter.KEYWORD_ONLY)]
```

Comment:
Sorry if this review is premature but shouldn't it be `Parameter('spec', Parameter.POSITIONAL_OR_KEYWORD)`. Also not that it matters much but why separate lists for `cls` and `spec`?

Reply:
Both good suggestions that I'll implement in the cleanup pass once I finish getting the tab completion working elsewhere.

Reply:
Addressed in 91b6359